src/Microsoft.ComponentDetection.Contracts/TypedComponent/TypedComponent.cs

```diff
@@ -30,6 +30,10 @@ internal TypedComponent()
     [JsonIgnore]
     internal string DebuggerDisplay => $"{this.Id}";
 
+    [field: JsonIgnore]
+    [JsonIgnore]
+    protected string CacheId { get; set; }
```

Comment:
Is there a different pattern for the caching that will make it more obvious for future TypedComponents to utilize in the future? Maybe make ComputeId() an abstract method and then encapsulate the caching into a getter property here in the base class.

Reply:
Good suggestion.